src/ape/api/providers.py

```diff
@@ -973,6 +992,75 @@ def get_transactions_by_block(self, block_id: BlockID) -> Iterator:
         for transaction in block.get("transactions", []):
             yield self.network.ecosystem.create_transaction(**transaction)
 
+    def get_transactions_by_account_nonce(
+        self,
+        account: AddressType,
+        start_nonce: int,
+        stop_nonce: int,
+    ) -> Iterator[ReceiptAPI]:
+        if self.network.name != "local" and (stop_nonce - start_nonce) > 2:
+            # NOTE: RPC usage might be acceptable to find 1 or 2 transactions reasonably quickly
+            logger.warning(
+                "Performing this action is likely to be very slow and may "
+                f"use {20 * (stop_nonce - start_nonce)} or more RPC calls. "
+                "Consider installing an alternative data query provider plugin."
+            )
+
+        if start_nonce > stop_nonce:
+            raise ValueError("Starting nonce cannot be greater than stop nonce for search")
```

Comment:
Super small semantic thing but should the order of these conditions be switched? If `start_nonce > stop_nonce` then the warning would say "use -20 or more RPC calls" for instance

Reply:
That's a great point